### PR TITLE
Rearrange firewall rules form and add help text

### PR DESCRIPTION
**Pull request by david-crespo**
The diff is a nightmare compared to the amount of changes because it's a lot of shuffling and fiddly margin stuff. I'd focus on checking the rendered result.

https://console-git-firewall-rules-help-oxidecomputer.vercel.app/projects/mock-project/vpcs/mock-vpc

image

image
﻿
image

### CODE REVIEW

**Comment by benjaminleonard** [comment]
Some of those notes could possible be moved to the help text / description directly on the input.

Whilst you're in here, some placeholder values would be wonderful.

File: app/forms/firewall-rules-create.tsx [hunk 1]
```diff
@@ -182,8 +215,6 @@ export const CommonFields = ({ error, control }: CommonFieldsProps) => {
       <NameField name="name" control={control} />
       <DescriptionField name="description" control={control} />
 
-      <FormDivider />
-
       <NumberField
         name="priority"
```
Comment (benjaminleonard) [inline app/forms/firewall-rules-create.tsx:219]: Any thoughts about moving this to the end? I think having it first gives it undue importance.

Comment (david-crespo) [reply app/forms/firewall-rules-create.tsx]: Moved it to the bottom of the top section.

**Comment by david-crespo** [comment]
I don't think I can squeeze the help into individual fields because it's really about the whole thing. Placeholders are also kind of tough because they depend on which type you have picked in the listbox.

**Comment by david-crespo** [comment]
That's a cool idea. I'll see if there's a way to make it work.

**Comment by david-crespo** [comment]
What you might be thinking of that came up before was making Enter submit the mini form when you're in one of those fields rather than submitting the whole form, we fixed that.

**Comment by david-crespo** [comment]
I don't think I can make it make sense to block leftover stuff in the input. You wouldn't want to just disable submit, you'd want it to be a validation error you can go find, but you'd only want it to be triggered by submit. We'd have to get pretty elaborate and I don't think it's worth it.

I added validation for port range because it's easy, but for the target and host fields, it's a lot more complicated because the way you validate the value input depends on which type is selected in the listbox. That's captured in #2187

**Comment by david-crespo** [comment]
I'm going to merge this as-is and we can redo the text boxes in a different style as a followup per design feedback.

**Pull request merged**